bundles/mapping/mapmodule/plugin/layers/LayerSelectionPopup.js

```diff
@@ -0,0 +1,123 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { showPopup } from 'oskari-ui/components/window';
+import { MetadataIcon } from 'oskari-ui/components/icons';
+import styled from 'styled-components';
+import { Checkbox, Message, Radio } from 'oskari-ui';
+
+const BUNDLE_NAME = 'MapModule';
+
+const POPUP_ID = 'LayerSelection';
+
+const THEME_LIGHT = {
+    color: {
+        primary: '#fffbf3',
+        accent: '#ffd622'
+    }
+};
+
+const THEME_DARK = {
+    color: {
+        primary: '#47484c',
+        accent: '#ffd622'
+    }
+};
+
+const Content = styled('div')`
+    margin: 12px 24px 24px;
+`;
+
+const LayerRow = styled('div')`
+    display: flex;
+    flex-direction: row;
+    justify-content: space-between;
+    align-items: center;
+`;
+
+const RadioGroup = styled(Radio.Group)`
+    margin-bottom: 20px;
+`;
+
+const StyledCheckbox = styled(Checkbox)`
+    margin-right: 5px;
+`;
+
+const LayerSelectionPopup = ({ baseLayers, layers, showMetadata, setLayerVisibility }) => {
+    const normalLayers = layers.filter(layer => baseLayers.findIndex(bl => bl.getId() === layer.getId()) < 0);
+    return (
+        <Content>
+            <h3><Message bundleKey={BUNDLE_NAME} messageKey='plugin.LayerSelectionPlugin.chooseDefaultBaseLayer' /></h3>
+            <RadioGroup
+                value={baseLayers.find(l => l.isVisible())}
+                onChange={e => setLayerVisibility(e.target.value, e.target.checked, true)}
+            >
+                {baseLayers.map(layer => {
+                    return (
+                        <div key={layer.getId()}>
+                            <LayerRow>
+                                <Radio.Choice value={layer}>
+                                    {layer.getName()}
+                                </Radio.Choice>
+                                {showMetadata && (<MetadataIcon metadataId={layer.getMetadataIdentifier()} />)}
+                            </LayerRow>
+                        </div>
+                    );
+                })}
+            </RadioGroup>
+            <h3><Message bundleKey={BUNDLE_NAME} messageKey='plugin.LayerSelectionPlugin.chooseOtherLayers' /></h3>
+            {normalLayers.map(layer => {
+                return (
+                    <LayerRow key={layer.getId()}>
+                        <div>
+                            <StyledCheckbox
+                                checked={layer.isVisible()}
+                                onChange={e => setLayerVisibility(layer, e.target.checked, false)}
+                            />
+                            <span>{layer.getName()}</span>
+                        </div>
+                        {showMetadata && (<MetadataIcon metadataId={layer.getMetadataIdentifier()} />)}
```

Comment:
Just noticed that this new implementation is missing the styleChange option. We need to implement that as well.